File: octopoints_flutter/lib/db/entity/team_user_relation_entity.dart
import 'package:floor/floor.dart';
import 'package:octopoints_flutter/db/database.dart';

@Entity(tableName: 'teamUserRelation', primaryKeys: [
  'teamId',
  'userId'
], foreignKeys: [
  ForeignKey(
      childColumns: ['teamId'],
      parentColumns: ['id'],
      entity: TeamEntity,
      onDelete: ForeignKeyAction.cascade),
  ForeignKey(
      childColumns: ['userId'],
      parentColumns: ['id'],
      entity: UserEntity,
      onDelete: ForeignKeyAction.cascade)
])
class TeamUserRelationEntity {
  final int teamId;
  final int userId;

  TeamUserRelationEntity(
    this.teamId,
    this.userId,
  );
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/entity/TeamUserRelationEntity.kt
package dev.tpavani.octopoints.db.entity

import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(tableName = "teamUserRelation",
    primaryKeys = ["teamId", "userId"],
    foreignKeys = [
        ForeignKey(
            entity = teamEntity::class,
            parentColumns = ["id"],
            childColumns = ["teamId"],
            onDelete = ForeignKey.CASCADE,
        ),
        ForeignKey(
            entity = UserEntity::class,
            parentColumns = ["id"],
            childColumns = ["userId"],
            onDelete = ForeignKey.CASCADE,
        )
    ]
)
class TeamUserRelationEntity(
    val teamId: Int,
    val userId: Int,
)